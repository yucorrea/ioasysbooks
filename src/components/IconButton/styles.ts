import styled from 'styled-components/native';

export const Container = styled.TouchableOpacity`
  border: 1px solid rgba(51, 51, 51, 0.2);
  width: 32px;
  height: 32px;
  border-radius: 16px;
  justify-content: center;
  align-items: center;
`;
