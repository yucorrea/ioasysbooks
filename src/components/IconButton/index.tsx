import React from 'react';
import { TouchableOpacityProps } from 'react-native';
import { useTheme } from 'styled-components/native';

import Icon from 'react-native-vector-icons/Ionicons';

import { Container } from './styles';

interface Props extends TouchableOpacityProps {
  onPress(): void;
  icon: string;
}

export function IconButton({ onPress, icon, ...rest }: Props) {

  const { colors } = useTheme();

  return (
    <Container {...rest} onPress={onPress}>
      <Icon name={icon} size={16} color={colors.title} />
    </Container>
  );
}
