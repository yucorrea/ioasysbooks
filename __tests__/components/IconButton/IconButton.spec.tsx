import React from 'react';

import IconButton from '../../../src/components/IconButton';
import {
  matchSnapshotWithTheme,
  renderWithTheme,
} from '../../../src/helpers/functions/testing';

describe('IconButton Component', () => {

  it('matches snapshots', () => {
    const mockFunction = jest.fn();
    const snap = matchSnapshotWithTheme(<IconButton icon='arrow-back-outline' onPress={mockFunction}/>);
    expect(snap).toMatchSnapshot();
  });

  it('render correctly', () => {
    const mockFunction = jest.fn();
    renderWithTheme( <IconButton icon="arrow-back-outline" onPress={mockFunction} /> );
  });

});
