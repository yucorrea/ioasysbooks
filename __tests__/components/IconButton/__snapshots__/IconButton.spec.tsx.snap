// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Component matches snapshots 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "rgba(51, 51, 51, 0.2)",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
      "width": 32,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#333333",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï„¨
  </Text>
</View>
`;

exports[`IconButton Component matches snapshots 2`] = `undefined`;
